Hbo 3

NiakhaichykVitaliy opened the pull request.

NiakhaichykVitaliy committed:
[HBO-3] Create class Pokemon

NiakhaichykVitaliy committed:
[HBO-3]
Edit activity_main.xml
Create list_pokemon.xml

NiakhaichykVitaliy committed:
[HBO-3] Edit build.gradle.kts
Create PokemonAdapter
Create PokemonList
Create list_pokemon.xml
Edit activity_main.xml

DerevyagoKonstantin commented on app/src/main/java/com/example/sample/MainActivity.java:
```diff
@@ -6,14 +6,44 @@
+import java.util.List;
 
 public class MainActivity extends AppCompatActivity {
+    private RecyclerView recyclerView;
```
это всё должно быть в отдельном фрагменте, который будет на активити

DerevyagoKonstantin commented on app/src/main/java/com/example/sample/Pokemon.java:
```diff
@@ -0,0 +1,21 @@
+        this.imagePokemon = imagePokemon;
+    }
+
+    public String getNamePokemon() {
```
обычно в таких классах-модельках делают геттегры и сеттеры для всех параметров

DerevyagoKonstantin commented on app/src/main/java/com/example/sample/PokemonAdapter.java:
```diff
@@ -0,0 +1,55 @@
+    private List<Pokemon> pokemonList;
+
+    public PokemonAdapter(List<Pokemon> pokemonList) {
+        this.pokemonList = pokemonList;
```
список лучше передавать отдельной функцией, а не в конструкторе. иначе тебе надо будет заново создавать весь адаптер при изменении списка

DerevyagoKonstantin commented on app/src/main/java/com/example/sample/PokemonAdapter.java:
```diff
@@ -0,0 +1,55 @@
+    @Override
+    public void onBindViewHolder(@NonNull PokemonAdapter.PokemonViewHolder holder, int position) {
+        Pokemon pokemon = pokemonList.get(position);
+        holder.pokemonName.setText(pokemon.getNamePokemon());
```
давай лучше во вьюхолдере сделаем функцию bind, в которой будут сеттиться все параметры

DerevyagoKonstantin commented on app/src/main/res/layout/list_pokemon.xml:
это скорее pokemon_item

DerevyagoKonstantin commented on app/src/main/res/layout/list_pokemon.xml:
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
зачем тут линер?

DerevyagoKonstantin commented on app/src/main/res/layout/list_pokemon.xml:
```diff
@@ -0,0 +1,32 @@
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    tools:context=".MainActivity">
```
при чем тут мейн активити?

DerevyagoKonstantin commented on app/src/main/res/layout/list_pokemon.xml:
```diff
@@ -0,0 +1,32 @@
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+
+        <RelativeLayout
```
зачем релатив?

DerevyagoKonstantin commented on app/src/main/res/values/dimens.xml:
```diff
@@ -6,5 +6,5 @@
     <dimen name="corner_radius">15dp</dimen>
     <dimen name="rouned_corner_width">1dp</dimen>
-
+    <dimen name="image_size">80dp</dimen>
```
лучше сразу указать какой именно картинки. например, pokemon_image_size или типо того

DerevyagoKonstantin commented on app/src/main/res/values/strings.xml:
```diff
@@ -6,5 +6,26 @@
     <string name="login_password_error">Please enter password</string>
+    <string name="title_list_pokemon">Pokemon</string>
+    <string name="picachu">Picachu</string>
+    <string name="picachu_image">https://www.film.ru/sites/default/files/images/10(186).jpg</string>
```
урлы нет смысла выносить в ресурсы, если только ты не собираешься их как то менять для разных языков/стран/размеров девайса и т.д.

в принципе мы потом будем брать их из запроса, так что давай лучше урлы и имена просто в классе обычными строчками

DerevyagoKonstantin commented on gradle/libs.versions.toml:
```diff
@@ -1,21 +1,33 @@
 material = "1.12.0"
 activity = "1.9.3"
 constraintlayout = "2.2.0"
+playServicesAdsLite = "23.5.0"
```
а зачем тебе рекламные либы?

NiakhaichykVitaliy committed:
[HBO-3]
Add getter and setter in Pokemon class

NiakhaichykVitaliy committed:
[HBO-3]
Edit libs.versions.toml

NiakhaichykVitaliy committed:
[HBO-3]
Edit list_pokemon.xml
Delete name and images url from res
Rename dimen name

NiakhaichykVitaliy committed:
[HBO-3]
Edit PokemonAdapter
Add method bind on PokemonViewHolder
Add method setPokemonList on PokemonViewHolder

NiakhaichykVitaliy committed:
[HBO-3]
Create PokemonFragment
Moved recycler to PokemonFragment
Create fragment_pokemon.xml

NiakhaichykVitaliy committed:
[HBO-3]
Edit width in FrameLayout

NiakhaichykVitaliy committed:
[HBO-3]
Import true widget.Toolbar
Delete trash constructor

NiakhaichykVitaliy committed:
[HBO-3]
Delete import Toolbar from MainActivity
Delete empty string from PokemonFragment

DerevyagoKonstantin commented on app/src/main/java/com/example/sample/MainActivity.java:
```diff
@@ -14,6 +15,11 @@ protected void onCreate(Bundle savedInstanceState) {
         EdgeToEdge.enable(this);
         setContentView(R.layout.activity_main);
+
+        PokemonFragment pokemonFragment = new PokemonFragment();
```
тут не ошибка, но лучше привыкай создавать фрагменты через статический метод (например getInstance(). вот типо как для активити startActivity()), чтобы случайно не передать какие нибудь аргументы в конструктор фрагмента, если они будут нужны (аргументы надо засовывать в бандл, чтобы они не потерялись при пересоздании, и про это легко забыть. а при вызове статического метода ты скорее всего про это вспомнишь)

DerevyagoKonstantin commented on app/src/main/java/com/example/sample/MainActivity.java:
```diff
@@ -14,6 +15,11 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_main);
+
+        PokemonFragment pokemonFragment = new PokemonFragment();
+        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
```
тоже не ошибка, но вроде бы эти функции транзакции возвращают тоже транзакцию и можно цепочкой сделать:
```
getSupportFragmentManager().beginTransaction()
    .replace(R.id.fragment_container, PokemonFragment.getInstance())
    .commit();
```

DerevyagoKonstantin commented on app/src/main/java/com/example/sample/PokemonAdapter.java:
```diff
@@ -0,0 +1,60 @@
+        notifyDataSetChanged();
+    }
+
+    public static class PokemonViewHolder extends RecyclerView.ViewHolder {
```
лучше наверное вынести внутренний класс вниз, чтобы не мешать с функциями

DerevyagoKonstantin commented on app/src/main/java/com/example/sample/PokemonAdapter.java:
```diff
@@ -0,0 +1,60 @@
+        notifyDataSetChanged();
+    }
+
+    public static class PokemonViewHolder extends RecyclerView.ViewHolder {
```
а почему статик?

NiakhaichykVitaliy replied:
статический класс не хранит ссылку на свои экземпляры класса, что не приводит к утечке памяти

DerevyagoKonstantin commented on app/src/main/java/com/example/sample/PokemonFragment.java:
```diff
@@ -0,0 +1,59 @@
+public class PokemonFragment extends Fragment {
+    private RecyclerView recyclerView;
+    private PokemonAdapter adapter;
+    private List<Pokemon> pokemonList;
```
кажется список не надо делать глобальной переменной класса. можно просто в функции создать

DerevyagoKonstantin commented on app/src/main/java/com/example/sample/Pokemon.java:
```diff
@@ -0,0 +1,37 @@
+
+public class Pokemon {
+    private int id;
+    private String namePokemon;
```
тоже не ошибка, но нет смысла добавлять `Pokemon` к параметрам внутри класса `Pokemon` - это и так понятно)

NiakhaichykVitaliy committed:
[HBO-3]
Rename vars in class Pokemon
Replace inner class to end
Delete global var pokemonList and add he's in method onCreateView

NiakhaichykVitaliy committed:
[HBO-3]
Create singleton for PokemonFragment
Refactor FragmentTransaction

DerevyagoKonstantin commented on app/src/main/java/com/example/sample/MainActivity.java:
```diff
@@ -14,6 +15,11 @@ protected void onCreate(Bundle savedInstanceState) {
+        PokemonFragment pokemonFragment = PokemonFragment.getInstance();
+        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction()
+                .replace(R.id.fragment_container, pokemonFragment);
+        fragmentTransaction.commit();
```
А чего коммит отдельной строчкой? Реплейс вроде бы тоже объект транзакции возвращает

DerevyagoKonstantin commented on app/src/main/java/com/example/sample/PokemonAdapter.java:
```diff
@@ -0,0 +1,60 @@
+        return pokemonList.size();
+    }
+
+    public static class PokemonViewHolder extends RecyclerView.ViewHolder {
```
Я может уже джаву забыл, но разве Статик для класса тут нужен?

DerevyagoKonstantin commented on app/src/main/res/layout/fragment_pokemon.xml:
```diff
@@ -0,0 +1,19 @@
+        android:id="@+id/toolbar"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/title_list_pokemon"
```
Если я правильно помню, то тут надо просто text на title поменять и тогда не надо будет в классе сетать строку

NiakhaichykVitaliy committed:
[HBO-3]
Refactor FragmentTransaction
Refactor Toolbar title

NiakhaichykVitaliy committed:
[HBO-3]
Delete import FragmentTransaction

NiakhaichykVitaliy committed:
[HBO-3]
Refactor method getInstance
